baseplate/clients/thrift.py

```diff
@@ -213,38 +214,38 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
                         prot.trans.set_header(b"Edge-Request", edge_context)
 
                     result = method(*args, **kwargs)
-            except TTransportException as exc:
-                # the connection failed for some reason, retry if able
-                span.finish(exc_info=sys.exc_info())
-                last_error = str(exc)
-                if exc.inner is not None:
-                    last_error += f" ({exc.inner})"
-                continue
-            except (TApplicationException, TProtocolException):
-                # these are subclasses of TException but aren't ones that
-                # should be expected in the protocol. this is an error!
-                span.finish(exc_info=sys.exc_info())
-                raise
-            except Error as exc:
-                # a 5xx error is an unexpected exception but not 5xx are
-                # not.
-                if 500 <= exc.code < 600:
+                except TTransportException as exc:
+                    # the connection failed for some reason, retry if able
+                    span.finish(exc_info=sys.exc_info())
```

Comment:
does `span.finish` tag this info on any of the metrics to understand why a call failed and was retried?

Reply:
Yes, [it sets `success` and `exception_type`](https://github.com/reddit/baseplate.py/blob/022e020eeeaada76efae75599137cf63bb1bee32/baseplate/observers/prometheus.py#L232-L236)